pkg/runctl/controller.go

```diff
@@ -277,14 +277,32 @@ func (c *Controller) syncHandler(key string) error {
 		c.changeState(pipelineRun, api.StateCleaning)
 	}
 
-	runManager := c.createRunManager(pipelineRun, pipelineRunsConfig)
-
 	if pipelineRun.GetStatus().State == api.StateUndefined {
 		if err = c.changeState(pipelineRun, api.StatePreparing); err != nil {
 			return err
 		}
 		c.metrics.CountStart()
 	}
+
+	// the configuration should be loaded once per sync to avoid inconsistencies
+	// in case of concurrent configuration changes
+	pipelineRunsConfig, err := c.getPipelineRunsConfig()
+	if err != nil {
+		if serrors.IsRecoverable(err) {
+			return err
+		}
```

Comment:
Shouldn't we log the error here? How do operators become aware of invalid controller configuration?

Reply:
I added an event as this should also be easier to monitor: See efa8b80